mesonbuild/modules/hotdoc.py

```diff
@@ -370,28 +392,29 @@ def make_targets(self):
 
 
 class HotdocTargetHolder(_CustomTargetHolder['HotdocTarget']):
-    def __init__(self, target: HotdocTarget, interp):
+    def __init__(self, target: HotdocTarget, interp: Interpreter):
         super().__init__(target, interp)
         self.methods.update({'config_path': self.config_path_method})
 
     @noPosargs
     @noKwargs
-    def config_path_method(self, *args, **kwargs):
+    def config_path_method(self, *args: T.Any, **kwargs: T.Any) -> str:
         conf = self.held_object.hotdoc_conf.absolute_path(self.interpreter.environment.source_dir,
                                                           self.interpreter.environment.build_dir)
         return conf
 
 
-class HotdocTarget(build.CustomTarget):
-    def __init__(self, name, subdir, subproject, hotdoc_conf, extra_extension_paths, extra_assets,
-                 subprojects, environment, **kwargs):
+class HotdocTarget(CustomTarget):
+    def __init__(self, name: str, subdir: str, subproject: str, hotdoc_conf: File,
```

Comment:
Should the subproject type be SubProject?

Reply:
Sort of yes, sort of no. There are preexisting build.py issues with that.

We only need that specificity for cases where we use a FeatureCheck and pass that in, or cases where we pass the subproject on to an object that *does* use FeatureCheck. And that is, admittedly, the case for `build.Target` (but not its subclasses :scream:).